libusb/os/windows_winusb.c

```diff
@@ -2754,8 +2782,36 @@ static int winusbx_release_interface(int sub_api, struct libusb_device_handle *d
 	if (!HANDLE_VALID(winusb_handle))
 		return LIBUSB_ERROR_NOT_FOUND;
 
+	int initialized_iface;
+	if (priv->usb_interface[iface].is_associated_interface)	{
+		initialized_iface = priv->usb_interface[iface].first_associated_interface;
+	} else if (is_using_usbccgp){
+		initialized_iface = iface;
+	} else {
+		initialized_iface = 0;
+	}
+
+	usbi_dbg(ctx, "\tlibusb: release_interface(%p, %d) initialized_iface=%d claimed_count = %d\n", dev_handle, iface, initialized_iface, handle_priv->interface_handle[initialized_iface].claimed_interfaces_count);
+
 	WinUSBX[sub_api].Free(winusb_handle);
 	handle_priv->interface_handle[iface].api_handle = INVALID_HANDLE_VALUE;
+	handle_priv->interface_handle[initialized_iface].claimed_interfaces_count--;
+
+	if (handle_priv->interface_handle[initialized_iface].claimed_interfaces_count == 0)	{
+		usbi_info(ctx, "closing device handle '%s'\n", priv->path);
```

Comment:
Does this need some kind of locking in case two threads release interfaces at the same time?

Reply:
Probably.. could you share a bit more information (or direction where to look) about locking inside libusb? There is already `usbi_mutex_lock(&dev_handle->lock);` inside `libusb_release_interface` so it seems that things are already synchronized?

Reply:
Alright, this should be fine then.